windows/inc/leatherman/windows/scoped_resource.hpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file
+ * Declares the base class for scoped resources.
+ */
+#pragma once
+
+#include <functional>
+
+namespace leatherman { namespace windows {
+    /**
```

Comment:
Yeah, I think `scoped_resource` isn't windows specific, but of broad utility. If so, let's see if we can pull up into a non-windows util header somewhere.